### PR TITLE
Check if subject exists for recent subjects on homepage

### PR DESCRIPTION
**Pull request by goplayoutside3**
## Package
app-project

## Linked Issue and/or Talk Post
Closes: https://github.com/zooniverse/front-end-monorepo/issues/5765

## Describe your changes

- Check if a subject exists in `fetchSubjectData` helper before returning subjects array to RecentSubjects component
- Adjust error handling so that if a subject isn't found, the other recently commented-on subjects will still display

## How to Review

See in-code comment below about how to review by running locally.

Gravity Spy's home page was previously crashing due to the recently commented-on subjects, but enough comments have come through Talk that it's no longer crashing.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [x] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [x] Can submit a classification
- [x] Can sign-in and sign-out

## Bug Fix
- [x] The PR creator has listed user actions to use when testing if bug is fixed
- [x] The bug is fixed
- [ ] Unit tests are added or updated

### CODE REVIEW

**Comment by goplayoutside3** [comment]
Uh oh build failing, one sec!

**Comment by goplayoutside3** [comment]
Okay all good and ready for review now 👍

File: packages/app-project/src/screens/ProjectHomePage/components/ZooniverseTalk/components/RecentSubjects/helpers/fetchRecentSubjects/fetchSubjectData.js [hunk 1]
```diff
@@ -1,11 +1,26 @@
...
+      id: subjectIds.join(',')
+    })
+    const { subjects } = response.body
+    // response does not come back in the same order as subjectIds array
+    // so we have to order them correctly here
+    recentlyDiscussedSubjects = subjectIds.reduce((acc, id) => {
+      // sometimes old subjects have been deleted, but their Talk thread still exists via their subject.id
+      if (subjects.find(subject => subject.id === id)) {
+        acc.push(subjects.find(subject => subject.id === id))
+      }
```
Comment (mcbouslog) [inline packages/app-project/src/screens/ProjectHomePage/components/ZooniverseTalk/components/RecentSubjects/helpers/fetchRecentSubjects/fetchSubjectData.js:17]: Optionally, could do the `subjects.find` once and reuse result:
```suggestion
      const subject = subjects.find(subject => subject.id === id);
      if (subject) {
        acc.push(subject);
      }
```

**Review by mcbouslog — APPROVED** [review]
Confirmed fix locally, thank you for comment to test that was very helpful!

LGTM 👍

File: packages/app-project/src/screens/ProjectHomePage/components/ZooniverseTalk/components/RecentSubjects/helpers/fetchRecentSubjects/fetchSubjectData.js (hunk 1 above)
Comment (goplayoutside3) [reply packages/app-project/src/screens/ProjectHomePage/components/ZooniverseTalk/components/RecentSubjects/helpers/fetchRecentSubjects/fetchSubjectData.js:17]: Good point!

**Pull request merged**